tap_hubspot/__init__.py

```diff
@@ -307,8 +322,70 @@ def lift_properties_and_versions(record):
             record['properties_versions'] += versions
     return record
 
+def post_search_endpoint(url, data, params=None):
+    params = params or {}
+    hapikey = CONFIG['hapikey']
+    if hapikey is None:
+        if CONFIG['token_expires'] is None or CONFIG['token_expires'] < datetime.datetime.utcnow():
+            acquire_access_token_from_refresh_token()
+        headers = {'Authorization': 'Bearer {}'.format(CONFIG['access_token'])}
+    else:
+        params['hapikey'] = hapikey
+        headers = {}
+
+    if 'user_agent' in CONFIG:
+        headers['User-Agent'] = CONFIG['user_agent']
+
+    headers['content-type'] = "application/json"
+
+    with metrics.http_request_timer(url) as timer:
+        resp = requests.post(
+            url=url,
+            json=data,
+            params=params,
+            headers=headers
+        )
+
+        resp.raise_for_status()
+
+    return resp
+
+def merge_responses(v1_data, v3_data):
+    for v1_record in v1_data:
+        v1_id = v1_record.get('dealId')
+        for v3_record in v3_data:
+            v3_id = v3_record.get('id')
+            if str(v1_id) == v3_id:
+                v1_record['properties'] = {**v1_record['properties'],
+                                           **v3_record['properties']}
+
+def process_v3_deals_records(v3_data):
+    """
+    This function:
+    1. filters out fields that don't contain 'hs_date_entered_*' and
+       'hs_date_exited_*'
+    2. changes a key value pair in `properties` to a key paired to an
+       object with a key 'value' and the original value
+    """
+    transformed_v3_data = []
+    for record in v3_data:
+        new_properties = {field_name : {'value': field_value}
+                          for field_name, field_value in record['properties'].items()
+                          if 'hs_date_entered' in field_name or 'hs_date_exited' in field_name or 'hs_time_in' in field_name}
```

Comment:
Again I think `any` could be used to reduce the number of `if` and `or` in this line...

Reply:
Done